### DISCUSSION TITLE
Full inference (without slicing) gives different results than YOLOv8

### DISCUSSION THREAD

**Discussion by andressrodrl** [obss/sahi#945] [Q&A] [answered]
I am using the same weights, confidence threshold, IOU treshold  and size, also choosing 'NMS' as post-process method in SAHI. But the results are different and the detected objects are lower on SAHI. Any clue?

EDIT: looking it closely, the predictions are actually the same, the problem is that SAHI doesnt actually uses the `model_confidence_threshold` im defining, so it doesnt detect as much. It seems it uses 0.25 as threshold even tho i set it to 0.01

**Comment by wuyifan2233** [comment]
> I am using the same weights, confidence threshold, IOU treshold and size, also choosing 'NMS' as post-process method in SAHI. But the results are different and the detected objects are lower on SAHI. Any clue?
> 
> EDIT: looking it closely, the predictions are actually the same, the problem is that SAHI doesnt actually uses the `model_confidence_threshold` im defining, so it doesnt detect as much. It seems it uses 0.25 as threshold even tho i set it to 0.01

Hi, have you solved this problem? I face the same situation with you.

**Comment by GCGeo** [comment]
It appears certain configuration parameters for the detector are only applied when sliced prediction is on (no_sliced_prediction = False). 
I had a similar problem with changing the confidence threshold, and also when trying to set class-agnostic NMS to True. The parameters are not passed on to the detector.  
I have tried passing the parameters directly to the batch predict() function, passing a config.yaml file with the parameters in the predict() function, and also creating a AutoDetectionModel with the set parameters and then passing the detection model to the predict() function, none of that worked.

**Comment by WaderLaken** [comment]
I also get an incomprehensible problem using SAHI with YOLOv8 model.

When I compute the metrics for val datesets using standard inference with a YOLOv8 Model, I could get **0.862 mAP50**. However, I just get **0.607 mAP50** using sliced inference with the same model, simultaneously setting "no_sliced_prediction=True" or setting "no_sliced_prediction=False" but tuning the "slice_size" to an oversized value, like 8000x8000, definitively inferring only one slice on a picture. By the way, I ensure the other reasoning variables the same. What a surprising result!

What's more, I make another try, using sliced inference with am appropriate slice-size (such as 4032x4032), and the **mAP50 rises to 0.931**. Now I confirm that SAHI is effective to improve the recognition accuracy, especially for tiny objects. However, I really wonder why the values between the last two experiments are markedly different?

**Comment by fcakyon** [comment, marked as answer]
there was a bug in YOLOv8 prediction thresholds. This PR fixed the issue: https://github.com/obss/sahi/pull/988

@andressrodrl @wuyifan2233 @WaderLaken @GCGeo
